frontend/javascripts/oxalis/view/left-border-tabs/mapping_settings_view.js

```diff
@@ -98,21 +120,15 @@ class MappingSettingsView extends React.Component<Props, State> {
       throw new Error("Invalid mapping type");
     }
 
-    const progressCallback = createProgressCallback({
-      pauseDelay: 500,
-      successMessageDelay: 2000,
-    });
-    Model.getLayerByName(this.props.layerName).setActiveMapping(
-      mappingName,
-      mappingType,
-      progressCallback,
-    );
+    message.loading({ content: "Activating Mapping", key: MAPPING_MESSAGE_KEY });
+
+    this.props.setMapping(this.props.layerName, mappingName, mappingType);
```

Comment:
I just had another thought: Maybe the `message` handling should be completely done by the sagas? I think, that would make even more sense, since the callers of `setMapping` don't need to take care of this. Imagine that we add "enable mapping" support to the context menu. Currently, that would require `message.loading(...)` in the context menu, too.

When moving the message-handling into the sagas, this component here would still need to know whether the JSON mapping is being downloaded. For that, the options would probably be:
- the current way (checking for `null`)
- the callback suggestion from above
- or having the saga store the information in the store

Reply:
Thanks for the great suggestions!

The reason why the saga is not taking care of the message-handling is that not on all mapping activations, a message is supposed to be shown (merger mode, dynamic mapping activation in the connectome viewer). However, similar to your proposal with the callback, this could be communicated via a parameter to the action :+1:

Regarding the loading indicator in the settings component, I think I'll remove that again. Originally, I introduced it, because I removed the antd notification, but since I re-added the notification, this loading indicator doesn't add too much value.